hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/managed/ManagedObject.java

```diff
@@ -42,7 +49,23 @@ public void close() {
 
   @Override
   protected void finalize() throws Throwable {
-    ManagedRocksObjectUtils.assertClosed(original);
+    ManagedRocksObjectUtils.assertClosed(this);
     super.finalize();
   }
+
+  public String getStackTrace() {
+    if (elements != null && elements.length > 0) {
+      // 15 lines should be good enough to know the caller of
+      // the unclosed instance?
+      int maxStackTraceLines = 15;
```

Comment:
This will only show up when something went wrong, it's okay not to limit the stacktrace or use a longer stack trace.

Reply:
Ok , I added this as the stackTrace caught in this example was approximately 50 lines and was occupying the entire screen and if there are multiple such instances it can flood the debug log with these but I guess we should be okay if there are not many such instances.